### DISCUSSION TITLE
Mention in the docs that you can't pass any prop to the "as" component in a Controller with the same name as the props the controller may have

### DISCUSSION THREAD

**Discussion by Sleepful** [react-hook-form/react-hook-form#1782] [Q&A]
Mention in the docs that you can't pass any prop to the "as" component in a Controller with the same name as the props the controller may have

Somewhere in the Get Started it mentions that props for "as" component get overridden if they have the same name as Controller props, however this is not mentioned in the API:
https://react-hook-form.com/api#Controller
*Furthermore*, it does not mention that even props *not* supplied to <Controller> but that <Controller> may have also override props in "As" component.

For instance, even if I am not passing a prop "onChange" to <Controller>, if I pass a prop "onChange" to "as" then it gets overridden by <Controller>'s default "onChange".

**Comment by bluebill1049** [comment]
can you explain that in code?

Controller `onChange` is made for the use case of a payload is not an event alike object, for example, react-select. 

```
onChange(e: ) {
  console.log(e) <-  it's an object instead of e.target.value
}
```

take a look this codesandbox for reference: https://codesandbox.io/s/react-hook-form-controller-079xx

**Comment by Sleepful** [comment]
For example:
```
    <Controller
      as={<SomeComponent onChange={()=>console.log("change")} />}
      control={control}
      name={name}
      rules={inputRef}
      value={value}
    />
```
inside `SomeComponent` the `onChange` prop will be replaced by the `Controller`'s `onChange`'s default prop, even though it wasn't passed to `Controller`. This means that if I do:

`console.log(props.onChange.toString())` inside `SomeComponent` I get this output:

```
(event) => {
        const data = commonTask(event);
        setValue(name, data, shouldValidate());
    }
```

the same from [this file](https://github.com/xiaoyu-tamu/react-hook-form/blob/3d2605feeb47202762735f0e3a111e96fea227a6/src/controller.tsx#L72)

The "Get Started" section mentions something about this in the "Controller" part:
https://react-hook-form.com/get-started/#ControlledInput
where it says:
```
      {/* Option 2: pass a JSX element to the Controller. */}
      {/* Note that any prop passed to the element will be overriden. */}
      {/* In this case, "SomeName" will be changed to "MyCheckbox". */}
      <Controller
         as={<Checkbox name="SomeName"/>}
         name="MyCheckbox"
         control={control}
         defaultValue={false}
      />
```

but I don't think it's clear enough about props that `Controller` may have but aren't required. Additionally the documentation doesn't mention anything about props being overridden:
https://react-hook-form.com/api/#Controller

**Reply by bluebill1049** [reply]
it's open source as well, feel free to contribute. The next version will have `render` props, so things maybe more clear for you.

**Reply by Sleepful** [reply]
Maybe I will, first I open an issue in case I misunderstood something about how it works.

I am also confused about the `value="test"` prop found in the "Get Started" section of the Controller but not documented in the API?

**Reply by bluebill1049** [reply]
good pick up, that's a mistake.

**Reply by bluebill1049** [reply]
i will amend it now.

**Reply by bluebill1049** [reply]
oh wait, that's for the value of that checkbox, if it's checked.

`<input type="checkbox" value="test" defaultChecked={true} />`

but i will still remove it, cause it will cause confusion for users who is not familiar with it.
